src/main/java/org/folio/edge/sip2/MainVerticle.java

```diff
@@ -103,148 +108,190 @@ public void start(Promise<Void> startFuture) {
 
     callAdminHealthCheckService();
 
-    // initialize the TokenCache
+    // Initialize the TokenCache
     TokenCacheFactory.initialize(config()
         .getInteger(SYS_TOKEN_CACHE_CAPACITY, DEFAULT_TOKEN_CACHE_CAPACITY));
 
-
-    // We need to reduce the complexity of this method...
     setupHanlders();
 
-    //set Config object's defaults
-    int port = config().getInteger("port"); // move port to netServerOptions
-    NetServerOptions options = new NetServerOptions(
-        config().getJsonObject("netServerOptions", new JsonObject()))
-        .setPort(port);
-
-    server = vertx.createNetServer(options);
-
-    log.info("Deployed verticle at port {}", port);
-
-    final Metrics metrics = Metrics.getMetrics(port);
-    metricsMap.putIfAbsent(port, metrics);
-
-    server.connectHandler(socket -> {
-
-      String clientAddress = socket.remoteAddress().host();
-      ThreadContext.put(IPADDRESS, clientAddress);
-      JsonObject tenantConfig = TenantUtils.lookupTenantConfigForIPaddress(multiTenantConfig,
-          clientAddress);
-
-      final SessionData sessionData = SessionData.createSession(
-          tenantConfig.getString("tenant"),
-          tenantConfig.getString("fieldDelimiter", "|").charAt(0),
-          tenantConfig.getBoolean("errorDetectionEnabled", FALSE),
-          tenantConfig.getString("charset", "IBM850"));
-      final String messageDelimiter = tenantConfig.getString("messageDelimiter", "\r");
-
-      socket.handler(RecordParser.newDelimited(messageDelimiter, buffer -> {
-        final Timer.Sample sample = metrics.sample();
-
-        if (Objects.isNull(sessionData.getTenant())) {
-          log.error("No tenant configured for address: {}  message ignored.", clientAddress);
-          return;
-        }
-
-        final String messageString = buffer.getString(0, buffer.length(), sessionData.getCharset());
-
-        log.debug("Received message: {}", messageString);
-
-        Command command = UNKNOWN;
-
-        try {
-          final Parser parser = Parser.builder()
-              .delimiter(sessionData.getFieldDelimiter())
-              .charset(Charset.forName(sessionData.getCharset()))
-              .errorDetectionEnabled(sessionData.isErrorDetectionEnabled())
-              .timezone(sessionData.getTimeZone())
-              .build();
-
-          //parsing
-          final Message<Object> message = parser.parseMessage(messageString);
+    var portList = determinePorts();
+
+    if (!portList.isEmpty()) {
+      AtomicInteger remainingServers = new AtomicInteger(portList.size());
+
+      for (int i = 0; i < portList.size(); i++) {
```

Comment:
Intellj suggests using an enhanced for loop `for (int port : portList) ` which seems nice since you don't need `i` later.

Reply:
done